Fix context test timing

LeeDr opened the pull request:
## Summary

Fixes: https://github.com/elastic/kibana/issues/23464 and other context test failures.

There are several actions in the docTable and filterBar services that cause changes in Kibana but don't wait for them to complete (almost certainly because we don't know what to check to tell when the action is done).
Since these are some of the highest failing tests in CI I took the drastic action of just adding sleeps of about 2 seconds after those actions.  The tests now pass for me locally and I tested locally with the `Slow 3G` network setting and they still passed.
We can create a new issue to replace those sleeps with some better method once we figure out what that is.  But keep in mind, it won't save 2 seconds per call.  It will save some currently unknown portion of that 2 seconds per call.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

LeeDr commented:
retest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7271/)

LeeDr commented:
x-pack tests failed to even start (not related to this PR);
```
00:26:42.197    │ debg [chromedriver:stdout] Starting ChromeDriver 2.42.591071 (0b695ff80972cc1a65a5cd643186d2ae582cd4ac) on port 9515
00:26:42.197    │      Only local connections are allowed.
00:26:42.686    │ debg [chromedriver:ping] attempting to reach at "http://localhost:9515"
00:26:42.688    │ debg [chromedriver:ping] success
00:26:42.690    │ debg [leadfoot:command] Creating session
00:27:12.711    │ debg [leadfoot:command] Creating session
00:27:42.738    │ debg [leadfoot:command] Creating session
00:28:12.768    │ debg [leadfoot:command] Creating session
00:28:42.717    │ warn Failure loading service "remote"
00:28:42.720    │ERROR Error: remote failed to start within 2 minutes
00:28:42.720    │          at Promise.race (/var/lib/jenkins/workspace/elastic+kibana+pull-request+multijob-x-pack/kibana/test/functional/services/remote/leadfoot_command.js:60:13)
00:28:42.723    │          at <anonymous>
00:28:42.826    │ debg [leadfoot:command] Creating session
00:28:43.108    │ warn 1 processes left running, stop them with procs.stop(name): [ 'kibana' ]
00:28:43.108    │ debg Sending "undefined" to proc "kibana"
00:28:44.614    │ info [kibana] exited with null after 4 minutes
00:28:44.633 
00:28:44.633 Failure initializing 1 service(s)
```

retest

LeeDr commented:
retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7339/)

LeeDr commented:
Note that there is a method `await PageObjects.context.waitUntilContextLoadingHasFinished();` but apparently still doesn't wait for something to finish.  If we knew how to make the appropriate waitUntilItsFinished method we could replace the sleeps with it...  But I think there's a tradeoff of spending hours figuring that out and testing it, vs just adding a 2 second sleep.

LeeDr commented:
These tests have failed 3 Jenkins builds in the last 24 hours.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7355/)

LeeDr commented:
Same x-pack failure on oraclelinux (not related to this PR);
```
00:26:20.777    │ debg [chromedriver:stdout] Starting ChromeDriver 2.42.591071 (0b695ff80972cc1a65a5cd643186d2ae582cd4ac) on port 9515
00:26:20.777    │      Only local connections are allowed.
00:26:21.268    │ debg [chromedriver:ping] attempting to reach at "http://localhost:9515"
00:26:21.270    │ debg [chromedriver:ping] success
00:26:21.271    │ debg [leadfoot:command] Creating session
00:26:21.476    │ proc [kibana]   log   [16:26:55.672] [info][status][plugin:spaces@7.0.0-alpha1] Status changed from yellow to green - Ready
00:26:51.288    │ debg [leadfoot:command] Creating session
00:27:21.309    │ debg [leadfoot:command] Creating session
00:27:51.324    │ debg [leadfoot:command] Creating session
00:28:21.287    │ warn Failure loading service "remote"
00:28:21.287    │ERROR Error: remote failed to start within 2 minutes
00:28:21.287    │          at Promise.race (/var/lib/jenkins/workspace/elastic+kibana+pull-request+multijob-x-pack/kibana/test/functional/services/remote/leadfoot_command.js:60:13)
00:28:21.289    │          at <anonymous>
00:28:21.334    │ debg [leadfoot:command] Creating session
00:28:21.727    │ warn 1 processes left running, stop them with procs.stop(name): [ 'kibana' ]
00:28:21.727    │ debg Sending "undefined" to proc "kibana"
00:28:22.753    │ info [kibana] exited with null after 4 minutes
```

Bargs commented on test/functional/services/doc_table.js:
```diff
@@ -75,6 +77,8 @@ export function DocTableProvider({ getService }) {
       const rowExpandToggle = await this.getRowExpandToggle(row);
       await rowExpandToggle.click();
+      // Not sure how to determine this rowExpandToggle completed but the next step often fails without this sleep
+      await PageObjects.common.sleep(2222);
```
Why a sleep instead of wrapping the following line in a `retry.try`?

Bargs replied:
I don't think the click is failing, I think we're trying to find `detailsRow` too quickly after clicking. Those rows can be a little slow to render after the caret is clicked. So I'm proposing we wrap line 83 in a retry, e.g.

```
retry.try(async () => {
  return row.findByXpath('./following-sibling::*[@data-test-subj="docTableDetailsRow"]');
});
```

This would allow the test to wait a moment for the row to render, without necessarily having to wait a full two seconds, correct?

Stacey-Gammon replied:
You could push this into the find service with a new call `find.descendantByXpath(selector, parentElement)` . This should use then our best practices for the retry and making sure the element isn't stale.

Bargs commented on test/functional/services/filter_bar.js:
```diff
@@ -44,18 +45,24 @@ export function FilterBarProvider({ getService }) {
       const filterElement = await testSubjects.find(`filter & filter-key-${key}`);
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} removeFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
```
`filter-key-${key}` should no longer exist in the page, and the results should reload (global loading indicator will appear and then disappear when finished). Can we look for those indications?

LeeDr replied:
I can try that.

LeeDr replied:
I removed the sleep and added the more generic;
```
      await PageObjects.context.waitUntilContextLoadingHasFinished();
      await PageObjects.header.awaitGlobalLoadingIndicatorHidden();
```

Stacey-Gammon commented on test/functional/apps/context/index.js:
```diff
@@ -21,12 +21,14 @@ export default function ({ getService, getPageObjects, loadTestFile }) {
       await remote.setWindowSize(1200, 800);
       await esArchiver.loadIfNeeded('logstash_functional');
       await esArchiver.load('visualize');
+      await kibanaServer.uiSettings.replace({ 'dateFormat:tz': 'UTC', 'defaultIndex': 'logstash-*' });
```
I think you want to add the animation setting to off here.  It's possible that would fix it without the sleeps.  Though I know testing that theory will take forever....

Stacey-Gammon replied:
Oh also, I'm pretty sure `defaultIndex`: `logstash-*` does nothing because we need to use an id. this is used prolifically incorrectly.

LeeDr replied:
@Stacey-Gammon This test loads the `visualize` esArchiver data set which contains a logstash-* index pattern with that same id.  But it doesn't contain a config doc.  

```
{
  "type": "doc",
  "value": {
    "index": ".kibana",
    "type": "doc",
    "id": "index-pattern:logstash-*",
    "source": {
      "type": "index-pattern",
      "index-pattern": {
        "title": "logstash-*",
```

The way this has been working prior to this PR is that code in the navigateTo method in common_page checks if where we're navigating to and if it requires a default index pattern, and there isn't one, it sets it and logs some warnings.  (but now I see should also add 
`'telemetry:optIn': false` for cases where we run functional tests against a kibana with x-pack features running).

```
          return kibanaServer.uiSettings.getDefaultIndex()
            .then(function (defaultIndex) {
              if (appName === 'discover' || appName === 'visualize' || appName === 'dashboard') {
                if (!defaultIndex) {
                  // https://github.com/elastic/kibana/issues/7496
                  // Even though most tests are using esClient to set the default index, sometimes Kibana clobbers
                  // that change.  If we got here, fix it.
                  log.debug(' >>>>>>>> WARNING Navigating to [' + appName + '] with defaultIndex=' + defaultIndex);
                  log.debug(' >>>>>>>> Setting defaultIndex to "logstash-*""');
                  return kibanaServer.uiSettings.update({
                    'dateFormat:tz': 'UTC',
                    'defaultIndex': 'logstash-*',
                    'telemetry:optIn': false
                  });
                }
              }
            })
```

LeeDr replied:
You can see that WARNING logged when you run it locally.  It's a bit harder to find it when run in CI since it's debug logging.

Stacey-Gammon replied:
ahhhh, i understand now.  I saw that error in my tests but I'm using a different archive so logstash index is an id, not the name.  Good idea on the telemetry!  Still think it need the animations turned off too.

Bargs commented on test/functional/services/filter_bar.js:
```diff
@@ -44,18 +45,24 @@ export function FilterBarProvider({ getService }) {
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} disableFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
+      await PageObjects.common.sleep(2002);
```
The filter element should have a data-test-subj that goes from `filter-enabled` to `filter-disabled` and the global loading indicator should fire, can we look for those?

LeeDr replied:
I'm doubting whether checking the state of the filter pill is a good indication that the rest of the changes in the UI are complete.  It seems like that filter change would be the first thing that happens and everything else would still be changing?

The other issue here is that this method toggles, so to wait for that change I'd need to first get that state and then make the wait depend on the opposite state.  Not too hard, but not sure it's worth it (see above).

Or, I could change it so we have `enableFilter` and `disableFilter` methods.  Then the check that it has changed is simple.  I think that would simplify the test too.  Right now when you read the test and it says toggleFilterEnabled you don't know if it's enabling or disabling the filter.
What do you think ^?

Bargs replied:
> I'm doubting whether checking the state of the filter pill is a good indication that the rest of the changes in the UI are complete. It seems like that filter change would be the first thing that happens and everything else would still be changing?

Waiting just for the filter change definitely isn't sufficient, that's why I mentioned the global loading indicator as well.

> I could change it so we have enableFilter and disableFilter methods. Then the check that it has changed is simple.

I think that's probably a good idea.

LeeDr replied:
For now I've just added this and the tests have passed several times on Jenkins.
`await PageObjects.header.awaitGlobalLoadingIndicatorHidden();`

Bargs commented on test/functional/services/filter_bar.js:
```diff
@@ -44,18 +45,24 @@ export function FilterBarProvider({ getService }) {
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} pinFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
+      await PageObjects.common.sleep(2003);
```
The tooltip and aria-label on the button will change, we could look at those or add a data-test-subj. This does not cause a search reload so I'm surprised it would cause a race condition to occur.

LeeDr replied:
You're right that pinning/unpinning shouldn't cause a reload.  I just added the sleep in this case to be consistent across the actions.  I'll just remove this sleep and test a few times.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7384/)

LeeDr commented:
I started out wanting to just add a couple 2 second sleeps to fix these tests, but 5 hours later I've removed almost all the sleeps and have the context tests passing reliably for me locally with NO MORE RETRIES :-)
(The retries are still in the code but not getting hit in my local runs.)

I moved some waitUntilContextLoadingHasFinished() from the tests into the page_object and/or service methods.

The disableAnimations was already happening (I didn't look to figure out where).  I see it in the logging;
`applying update to kibana config: {"accessibility:disableAnimations":true}`

LeeDr commented:
My local results;
```
$ time node.exe scripts/functional_test_runner.js --bail --debug --grep 'context app' | while read line; do echo "[`date +%H:%M:%S.%N`] $line"; done
[17:33:21.691808700] debg Loading config file from "C:\Users\LeeDr\git\kibana\test\functional\config.js"
[17:33:22.320688500] debg Loading config file from "C:\Users\LeeDr\git\kibana\test\common\config.js"
[17:33:23.776091400] info Config loaded
[17:33:24.885779500] debg Starting local chromedriver at port 9515
[17:33:24.933590200] debg [chromedriver:ping] attempting to reach at "http://localhost:9515"
[17:33:24.980147700] debg [chromedriver:stdout] Starting ChromeDriver 2.42.591088 (7b2b2dca23cca0862f674758c9a3933e685c27d5) on port 9515
[17:33:25.034553100] Only local connections are allowed.
[17:33:25.400948600] debg [chromedriver:ping] success
[17:33:25.449182400] debg [leadfoot:command] Creating session
[17:33:27.130180200] debg [leadfoot:command] Registering session for teardown
[17:33:27.181517400] debg [leadfoot:command] Completing session capabilities
[17:33:30.725171600] info Remote initialized
[17:33:31.452748400] info Starting tests
[17:33:31.498620500]
[17:33:31.544621900] └-: context app
[17:33:31.605781000] └-> "before all" hook
[17:33:31.665046800] └-> "before all" hook
[17:33:31.717848500] │ info [logstash_functional] Loading "mappings.json"
[17:33:31.834017300] │ info [logstash_functional] Loading "data.json.gz"
[17:33:31.885136300] │ info [logstash_functional] Created index "logstash-2015.09.22"
[17:33:31.941489900] │ debg [logstash_functional] "logstash-2015.09.22" settings {"index":{"number_of_shards":"1","analysis":{"analyzer":{"url":{"type":"standard","max_token_length":"1000","tokenizer":"uax_url_email"}}},"number_of_replicas":"0"}}
[17:33:32.154479400] │ info [logstash_functional] Created index "logstash-2015.09.20"
[17:33:32.200946300] │ debg [logstash_functional] "logstash-2015.09.20" settings {"index":{"number_of_shards":"1","analysis":{"analyzer":{"url":{"type":"standard","max_token_length":"1000","tokenizer":"uax_url_email"}}},"number_of_replicas":"0"}}
[17:33:32.284616800] │ info [logstash_functional] Created index "logstash-2015.09.21"
[17:33:32.427987300] │ debg [logstash_functional] "logstash-2015.09.21" settings {"index":{"number_of_shards":"1","analysis":{"analyzer":{"url":{"type":"standard","max_token_length":"1000","tokenizer":"uax_url_email"}}},"number_of_replicas":"0"}}
[17:33:42.332617900] │ info [logstash_functional] Indexed 4633 docs into "logstash-2015.09.22"
[17:33:42.383547600] │ info [logstash_functional] Indexed 4757 docs into "logstash-2015.09.20"
[17:33:42.535498300] │ info [logstash_functional] Indexed 4614 docs into "logstash-2015.09.21"
[17:33:43.036780200] │ info [visualize] Loading "mappings.json"
[17:33:43.099137500] │ info [visualize] Loading "data.json.gz"
[17:33:43.271089400] │ info [visualize] Deleted existing index ".kibana_1"
[17:33:43.325960800] │ info [visualize] Deleted existing index ".kibana_2"
[17:33:43.540161300] │ info [visualize] Created index ".kibana"
[17:33:43.604113100] │ debg [visualize] ".kibana" settings {"index":{"number_of_shards":"1","number_of_replicas":"1"}}
[17:33:43.660798100] │ info [visualize] Indexed 3 docs into ".kibana"
[17:33:54.007224500] │ info [migrations] Creating index .kibana_2.
[17:33:54.248634300] │ info [migrations] Reindexing .kibana to .kibana_1
[17:33:54.845659300] │ info [migrations] Migrating .kibana_1 saved objects to .kibana_2
[17:33:54.991215700] │ debg [migrations] Migrating saved objects index-pattern:logstash-*, index-pattern:logstash*, visualization:Shared-Item-Visualization-AreaChart
[17:33:55.049711800] │ info [migrations] Pointing alias .kibana to .kibana_2.
[17:33:55.109013600] │ info [migrations] Finished in 1065ms.
[17:33:55.163186300] │ debg applying update to kibana config: {"accessibility:disableAnimations":true}
[17:33:59.255807200] │ debg replacing kibana config doc: {"dateFormat:tz":"UTC","defaultIndex":"logstash-*","telemetry:optIn":false}
[17:34:00.316010700] │ debg navigating to discover url: http://localhost:5620/app/kibana#/discover
[17:34:00.427100500] │ debg navigate to: http://localhost:5620/app/kibana#/discover
[17:34:00.856015800] │ debg ... sleep(700) start
[17:34:01.550780700] │ debg ... sleep(700) end
[17:34:01.594781400] │ debg returned from get, calling refresh
[17:34:02.169802700] │ debg currentUrl = http://localhost:5620/app/kibana#/discover
[17:34:02.215551300] │          appUrl = http://localhost:5620/app/kibana#/discover
[17:34:06.308848700] │ debg TestSubjects.find(kibanaChrome)
[17:34:06.362235700] │ debg findByCssSelector [data-test-subj~="kibanaChrome"]
[17:34:06.467403300] │ debg ... sleep(501) start
[17:34:06.969936300] │ debg ... sleep(501) end
[17:34:07.010703400] │ debg in navigateTo url = http://localhost:5620/app/kibana#/discover?_g=()
[17:34:07.058128400] │ debg TestSubjects.exists(statusPageContainer)
[17:34:07.112007300] │ debg existsByDisplayedByCssSelector [data-test-subj~="statusPageContainer"]
[17:34:09.552460200] └-: context link in discover
[17:34:09.603725900] └-> "before all" hook
[17:34:09.664564800] └-> "before all" hook
[17:34:09.725212900] │ debg navigating to discover url: http://localhost:5620/app/kibana#/discover
[17:34:09.780996700] │ debg navigate to: http://localhost:5620/app/kibana#/discover
[17:34:09.861371300] │ debg ... sleep(700) start
[17:34:10.565535700] │ debg ... sleep(700) end
[17:34:10.618729100] │ debg returned from get, calling refresh
[17:34:10.900604200] │ debg currentUrl = http://localhost:5620/app/kibana#/discover
[17:34:10.943716300] │          appUrl = http://localhost:5620/app/kibana#/discover
[17:34:14.661641600] │ debg TestSubjects.find(kibanaChrome)
[17:34:14.705055100] │ debg findByCssSelector [data-test-subj~="kibanaChrome"]
[17:34:14.805213300] │ debg ... sleep(501) start
[17:34:15.306823600] │ debg ... sleep(501) end
[17:34:15.347598200] │ debg in navigateTo url = http://localhost:5620/app/kibana#/discover?_g=()
[17:34:15.405330100] │ debg TestSubjects.exists(statusPageContainer)
[17:34:15.458790500] │ debg existsByDisplayedByCssSelector [data-test-subj~="statusPageContainer"]
[17:34:17.862420600] │ debg Setting absolute range to 2015-09-19 06:31:44.000 to 2015-09-23 18:31:44.000
[17:34:17.906902700] │ debg ensureTimePickerIsOpen
[17:34:17.953978600] │ debg TestSubjects.exists(timePicker)
[17:34:18.000111500] │ debg existsByDisplayedByCssSelector [data-test-subj~="timePicker"]
[17:34:20.422695900] │ debg TestSubjects.click(globalTimepickerButton)
[17:34:20.473826900] │ debg clickByCssSelector([data-test-subj~="globalTimepickerButton"])
[17:34:20.536003900] │ debg findByCssSelector [data-test-subj~="globalTimepickerButton"]
[17:34:20.814205800] │ debg TestSubjects.exists(timePicker)
[17:34:20.862415700] │ debg existsByDisplayedByCssSelector [data-test-subj~="timePicker"]
[17:34:20.911396800] │ debg --Clicking Absolute button
[17:34:20.948369400] │ debg showAbsoluteSection
[17:34:20.991147300] │ debg isAbsoluteSectionShowing
[17:34:21.041922100] │ debg doesCssSelectorExist input[ng-model='absolute.from']
[17:34:21.885333400] │ debg exists? false
[17:34:21.936056800] │ debg TestSubjects.click(timepicker-absolute-button)
[17:34:21.982147000] │ debg clickByCssSelector([data-test-subj~="timepicker-absolute-button"])
[17:34:22.039023200] │ debg findByCssSelector [data-test-subj~="timepicker-absolute-button"]
[17:34:22.213891600] │ debg getFromTime
[17:34:22.270256800] │ debg ensureTimePickerIsOpen
[17:34:22.327795600] │ debg TestSubjects.exists(timePicker)
[17:34:22.383170000] │ debg existsByDisplayedByCssSelector [data-test-subj~="timePicker"]
[17:34:22.434546700] │ debg showAbsoluteSection
[17:34:22.479571400] │ debg isAbsoluteSectionShowing
[17:34:22.530167000] │ debg doesCssSelectorExist input[ng-model='absolute.from']
[17:34:22.586265700] │ debg exists? true
[17:34:22.634992100] │ debg --Setting From Time : 2015-09-19 06:31:44.000
[17:34:22.676052300] │ debg setFromTime: 2015-09-19 06:31:44.000
[17:34:22.717211000] │ debg ensureTimePickerIsOpen
[17:34:22.772446000] │ debg TestSubjects.exists(timePicker)
[17:34:22.816468600] │ debg existsByDisplayedByCssSelector [data-test-subj~="timePicker"]
[17:34:22.873939600] │ debg showAbsoluteSection
[17:34:22.926426400] │ debg isAbsoluteSectionShowing
[17:34:22.978301000] │ debg doesCssSelectorExist input[ng-model='absolute.from']
[17:34:23.035773800] │ debg exists? true
[17:34:23.081999900] │ debg --Setting To Time : 2015-09-23 18:31:44.000
[17:34:23.135301600] │ debg setToTime: 2015-09-23 18:31:44.000
[17:34:23.188875100] │ debg ensureTimePickerIsOpen
[17:34:23.238835000] │ debg TestSubjects.exists(timePicker)
[17:34:23.292274300] │ debg existsByDisplayedByCssSelector [data-test-subj~="timePicker"]
[17:34:23.340127300] │ debg showAbsoluteSection
[17:34:23.393420200] │ debg isAbsoluteSectionShowing
[17:34:23.451292100] │ debg doesCssSelectorExist input[ng-model='absolute.from']
[17:34:23.501640300] │ debg exists? true
[17:34:23.666919600] │ debg clickGoButton
[17:34:23.721985700] │ debg TestSubjects.click(timepickerGoButton)
[17:34:23.778443400] │ debg clickByCssSelector([data-test-subj~="timepickerGoButton"])
[17:34:23.826040100] │ debg findByCssSelector [data-test-subj~="timepickerGoButton"]
[17:34:23.899604500] │ debg isGlobalLoadingIndicatorVisible
[17:34:23.963965700] │ debg TestSubjects.exists(globalLoadingIndicator)
[17:34:24.018872900] │ debg existsByDisplayedByCssSelector [data-test-subj~="globalLoadingIndicator"]
[17:34:24.075195800] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:24.121770200] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:24.171563000] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:25.542535800] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:25.599569900] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:25.664667000] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:25.716589300] │ debg TestSubjects.find(field-@message)
[17:34:25.764157400] │ debg findByCssSelector [data-test-subj~="field-@message"]
[17:34:25.823104900] │ debg TestSubjects.click(fieldToggle-@message)
[17:34:25.875481900] │ debg clickByCssSelector([data-test-subj~="fieldToggle-@message"])
[17:34:25.933194600] │ debg findByCssSelector [data-test-subj~="fieldToggle-@message"]
[17:34:26.216422700] │ debg TestSubjects.click(field-extension)
[17:34:26.265571200] │ debg clickByCssSelector([data-test-subj~="field-extension"])
[17:34:26.419132300] │ debg findByCssSelector [data-test-subj~="field-extension"]
[17:34:26.469190600] │ debg TestSubjects.click(field-geo.src)
[17:34:26.523369000] │ debg clickByCssSelector([data-test-subj~="field-geo.src"])
[17:34:26.566226400] │ debg findByCssSelector [data-test-subj~="field-geo.src"]
[17:34:27.132149500] │ debg isGlobalLoadingIndicatorVisible
[17:34:27.186329600] │ debg TestSubjects.exists(globalLoadingIndicator)
[17:34:27.238599100] │ debg existsByDisplayedByCssSelector [data-test-subj~="globalLoadingIndicator"]
[17:34:27.292868100] │ debg isGlobalLoadingIndicatorVisible
[17:34:27.345236000] │ debg TestSubjects.exists(globalLoadingIndicator)
[17:34:27.395338400] │ debg existsByDisplayedByCssSelector [data-test-subj~="globalLoadingIndicator"]
[17:34:27.456139200] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:27.504951000] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:27.571028700] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:27.620978400] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:27.674746900] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:27.728330700] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:28.115750400] └-> should open the context view with the selected document as anchor
[17:34:28.162204600] └-> "before each" hook: global before each
[17:34:28.213943100] │ debg TestSubjects.find(docTable)
[17:34:28.277863500] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:34:28.900322900] │ debg TestSubjects.find(docTable)
[17:34:28.948761200] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:34:29.618511900] └- √ pass
[17:34:29.681858900] └-> should open the context view with the same columns
[17:34:29.751505500] └-> "before each" hook: global before each
[17:34:29.801319700] │ debg TestSubjects.find(docTable)
[17:34:29.871374300] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:34:29.923392200] └- √ pass
[17:34:29.965247800] └-> should open the context view with the filters disabled
[17:34:30.014800100] └-> "before each" hook: global before each
[17:34:30.067456500] │ debg TestSubjects.exists(filter & filter-key-extension & filter-value-jpg & filter-disabled)
[17:34:30.133600200] │ debg existsByDisplayedByCssSelector [data-test-subj~="filter"][data-test-subj~="filter-key-extension"][data-test-subj~="filter-value-jpg"][data-test-subj~="filter-disabled"]
[17:34:30.191749000] │ debg TestSubjects.exists(filter & filter-key-geo.src & filter-value-IN & filter-disabled)
[17:34:30.251590000] │ debg existsByDisplayedByCssSelector [data-test-subj~="filter"][data-test-subj~="filter-key-geo.src"][data-test-subj~="filter-value-IN"][data-test-subj~="filter-disabled"]
[17:34:30.302950800] └- √ pass
[17:34:30.352748900] └-> "after all" hook
[17:34:30.392822200] └-: context filters
[17:34:30.443043600] └-> "before all" hook
[17:34:30.490592800] └-> should be addable via expanded doc table rows
[17:34:30.544095100] └-> "before each" hook: global before each
[17:34:30.590331700] └-> "before each" hook
[17:34:30.632479000] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:30.683913200] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:30.736907800] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:33.367096900] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:33.418516700] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:34.132698900] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:34.185902200] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:34.250711900] │ debg ... sleep(1000) start
[17:34:35.195396400] │ debg ... sleep(1000) end
[17:34:35.248379300] │ debg TestSubjects.find(docTable)
[17:34:35.308788000] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:34:35.562395600] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:35.618430700] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:35.672888700] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:35.712174100] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:35.757088100] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:35.806547300] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:35.856368000] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:35.988856700] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:36.055263100] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:36.107006200] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:36.154885300] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:36.205148800] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:36.257150700] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:36.313194400] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:36.496738600] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:36.566208800] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:36.631075200] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:36.692618000] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:36.827454400] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:36.879723600] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:36.956258600] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:37.009447000] │ debg TestSubjects.exists(filter & filter-key-geo.src & filter-value-IN & filter-enabled)
[17:34:37.075720100] │ debg existsByDisplayedByCssSelector [data-test-subj~="filter"][data-test-subj~="filter-key-geo.src"][data-test-subj~="filter-value-IN"][data-test-subj~="filter-enabled"]
[17:34:37.401948900] └- √ pass
[17:34:37.456898800] └-> should be toggleable via the filter bar
[17:34:37.506505400] └-> "before each" hook: global before each
[17:34:37.555967000] └-> "before each" hook
[17:34:37.743491600] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:37.790526100] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:37.839226900] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:40.322891300] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:40.366294700] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:41.087501800] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:41.142021100] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:41.191686200] │ debg ... sleep(1000) start
[17:34:42.131006700] │ debg ... sleep(1000) end
[17:34:42.181642600] │ debg TestSubjects.find(docTable)
[17:34:42.228409900] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:34:42.277168500] │ debg TestSubjects.click(addFilter)
[17:34:42.322744700] │ debg clickByCssSelector([data-test-subj~="addFilter"])
[17:34:42.372448200] │ debg findByCssSelector [data-test-subj~="addFilter"]
[17:34:42.569318900] │ debg TestSubjects.find(filterfieldSuggestionList)
[17:34:42.609527200] │ debg findByCssSelector [data-test-subj~="filterfieldSuggestionList"]
[17:34:42.990603100] │ debg TestSubjects.find(filterOperatorList)
[17:34:43.047729000] │ debg findByCssSelector [data-test-subj~="filterOperatorList"]
[17:34:43.375538900] │ debg TestSubjects.find(filterParams)
[17:34:43.423937100] │ debg findByCssSelector [data-test-subj~="filterParams"]
[17:34:43.559023500] │ debg TestSubjects.click(saveFilter)
[17:34:43.610907600] │ debg clickByCssSelector([data-test-subj~="saveFilter"])
[17:34:43.664266100] │ debg findByCssSelector [data-test-subj~="saveFilter"]
[17:34:43.785382100] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:43.834039200] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:43.883983200] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:43.939294100] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:43.993263000] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:44.054242900] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:44.119820600] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:44.222226100] │ debg TestSubjects.find(filter & filter-key-geo.src)
[17:34:44.269556200] │ debg findByCssSelector [data-test-subj~="filter"][data-test-subj~="filter-key-geo.src"]
[17:34:44.332826300] │ debg TestSubjects.click(filter & filter-key-geo.src disableFilter-geo.src)
[17:34:44.387166400] │ debg clickByCssSelector([data-test-subj~="filter"][data-test-subj~="filter-key-geo.src"] [data-test-subj~="disableFilter-geo.src"])
[17:34:44.454373300] │ debg findByCssSelector [data-test-subj~="filter"][data-test-subj~="filter-key-geo.src"] [data-test-subj~="disableFilter-geo.src"]
[17:34:44.503643500] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:44.568067400] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:44.615974800] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:44.666773900] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:44.722388800] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:44.767035100] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:44.818977800] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:44.913953900] │ debg TestSubjects.exists(filter & filter-key-geo.src & filter-value-IN & filter-disabled)
[17:34:44.970190800] │ debg existsByDisplayedByCssSelector [data-test-subj~="filter"][data-test-subj~="filter-key-geo.src"][data-test-subj~="filter-value-IN"][data-test-subj~="filter-disabled"]
[17:34:45.471326400] └- √ pass
[17:34:45.517297200] └-> "after all" hook
[17:34:45.592988700] └-: context size
[17:34:45.647297500] └-> "before all" hook
[17:34:45.741822600] └-> "before all" hook
[17:34:45.826070400] │ debg applying update to kibana config: {"context:defaultSize":"7","context:step":"3"}
[17:34:46.702187700] └-> should default to the `context:defaultSize` setting
[17:34:46.808785400] └-> "before each" hook: global before each
[17:34:47.001309700] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:47.099771500] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:47.174308100] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:49.513900100] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:49.553063600] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:50.453133100] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:50.509385500] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:50.565660100] │ debg ... sleep(1000) start
[17:34:51.494413400] │ debg ... sleep(1000) end
[17:34:51.565507100] │ debg TestSubjects.find(docTable)
[17:34:51.617751900] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:34:51.689666900] │ debg TestSubjects.find(predecessorCountPicker)
[17:34:51.739023200] │ debg findByCssSelector [data-test-subj~="predecessorCountPicker"]
[17:34:51.792583000] │ debg TestSubjects.find(successorCountPicker)
[17:34:51.844125800] │ debg findByCssSelector [data-test-subj~="successorCountPicker"]
[17:34:51.896063600] └- √ pass
[17:34:51.950912200] └-> should increase according to the `context:step` setting when clicking the `load newer` button
[17:34:52.008646700] └-> "before each" hook: global before each
[17:34:52.073264100] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:52.120777600] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:52.169190300] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:54.735436300] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:54.781647600] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:55.570026300] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:55.613954600] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:55.675509600] │ debg ... sleep(1000) start
[17:34:56.610828700] │ debg ... sleep(1000) end
[17:34:56.654750400] │ debg TestSubjects.find(docTable)
[17:34:56.719480700] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:34:56.795589800] │ debg Click Predecessor Load More Button
[17:34:56.865791000] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:56.917896200] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:56.974325600] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:57.023705100] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:34:57.082143800] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:34:57.134082300] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:34:57.184578500] │ debg isGlobalLoadingIndicatorVisible
[17:34:57.222058600] │ debg TestSubjects.exists(globalLoadingIndicator)
[17:34:57.294131900] │ debg existsByDisplayedByCssSelector [data-test-subj~="globalLoadingIndicator"]
[17:34:57.345290300] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:57.389639400] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:57.438234000] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:57.492738100] └- √ pass
[17:34:57.537170800] └-> should increase according to the `context:step` setting when clicking the `load older` button
[17:34:57.579821900] └-> "before each" hook: global before each
[17:34:57.644671200] │ debg awaitGlobalLoadingIndicatorHidden
[17:34:57.688500200] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:34:57.736335900] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:34:59.846175200] │ debg TestSubjects.find(successorLoadMoreButton)
[17:34:59.893398900] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:35:00.551996200] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:35:00.607159700] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:35:00.655757100] │ debg ... sleep(1000) start
[17:35:01.609782100] │ debg ... sleep(1000) end
[17:35:01.668136800] │ debg TestSubjects.find(docTable)
[17:35:01.744831500] │ debg findByCssSelector [data-test-subj~="docTable"]
[17:35:01.807966600] │ debg Click Successor Load More Button
[17:35:01.858640300] │ debg TestSubjects.find(successorLoadMoreButton)
[17:35:01.917050600] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:35:01.966365300] │ debg TestSubjects.find(successorLoadMoreButton)
[17:35:02.015428600] │ debg findByCssSelector [data-test-subj~="successorLoadMoreButton"]
[17:35:02.087118200] │ debg TestSubjects.find(predecessorLoadMoreButton)
[17:35:02.142416700] │ debg findByCssSelector [data-test-subj~="predecessorLoadMoreButton"]
[17:35:02.189739000] │ debg isGlobalLoadingIndicatorVisible
[17:35:02.236845300] │ debg TestSubjects.exists(globalLoadingIndicator)
[17:35:02.293097600] │ debg existsByDisplayedByCssSelector [data-test-subj~="globalLoadingIndicator"]
[17:35:02.361345000] │ debg awaitGlobalLoadingIndicatorHidden
[17:35:02.427287300] │ debg TestSubjects.find(globalLoadingIndicator-hidden)
[17:35:02.549145600] │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
[17:35:02.605498900] └- √ pass
[17:35:02.664026700] └-> "after all" hook
[17:35:02.719769900] └-> "after all" hook: unloadMakelogs
[17:35:02.857250500] │ info [logstash_functional] Unloading indices from "mappings.json"
[17:35:02.909789900] │ info [logstash_functional] Deleted existing index "logstash-2015.09.22"
[17:35:02.966331400] │ info [logstash_functional] Deleted existing index "logstash-2015.09.20"
[17:35:03.016512800] │ info [logstash_functional] Deleted existing index "logstash-2015.09.21"
[17:35:03.076278600] │ info [logstash_functional] Unloading indices from "data.json.gz"
[17:35:03.967440100] └-> "after all" hook
[17:35:04.085306400]
[17:35:04.132904400]
[17:35:04.183233600] 8 passing (1.0m)
[17:35:04.248560600]

real    1m47.717s
user    0m1.625s
sys     0m13.037s
```

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7428/)

LeeDr commented:
@Stacey-Gammon I think that logging is misleading.  I queried the config doc right when the test got past that point;
![image](https://user-images.githubusercontent.com/13542669/47923141-a4670900-de86-11e8-9887-3133b9f9ba02.png)

Stacey-Gammon commented:
Huh!  That is both very great, and also interesting because I thought that setting was being lost in some reporting tests and adding it back in fixed them.... Will have to investigate more to see if the setting is actually there or not.  cc @joelgriffith

LeeDr commented:
@Stacey-Gammon here; test\common\services\kibana_server\ui_settings.js is where the `replace` and an `update` methods are defined.  The `replace` method which is used in this case, first does a GET, then deletes any settings which are not overridden.  I think somehow this disableAnimations is one of those overridden things.  Then adds the new settings.

LeeDr commented:
Last run failed on this;
`✖ fail: "dashboard app using current data dashboard filtering adding a filter that excludes all data "before all" hook"`

retest

LeeDr commented:
retest

Stacey-Gammon commented:
awesome, thanks for the follow through @LeeDr, guess I can abandon https://github.com/elastic/kibana/pull/25054!

LeeDr commented:
Same as a previous run.  Context tests all passed but a failure on;
`✖ fail: "dashboard app using current data dashboard filtering adding a filter that excludes all data "before all" hook"`

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7505/)

LeeDr commented:
My changes broke some dashboard tests.  We can't call 
`      await PageObjects.context.waitUntilContextLoadingHasFinished();`
from the filter_bar service because it's used in other places like dashboard tests where they don't have those context buttons to wait for.  New commits coming soon...

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7543/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7577/)

LeeDr commented:
retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7578/)

LeeDr commented:
retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7619/)

LeeDr commented:
% PAsses in a row!  @Bargs OK with this?

Bargs commented:
LGTM

elasticmachine commented:
Pinging @elastic/kibana-app
